### DISCUSSION TITLE
Unable to enable ticklessmode

### DISCUSSION THREAD

**Discussion by tim-vu** [feilipu/Arduino_FreeRTOS_Library#82] [Q&A]
I have been trying for hours to enable tickless mode (https://www.freertos.org/low-power-tickless-rtos.html). At this point im not sure whether Im doing something wrong or there is a bug. 

```
#include <Arduino_FreeRTOS.h>

void TaskPrint( void *pvParameters );

void setup() {
  
  Serial.begin(9600);
  
  while (!Serial) {
  }

  xTaskCreate(
    TaskPrint
    ,  "Print"
    ,  128
    ,  NULL
    ,  2 
    ,  NULL );

}

#define configUSE_TICKLESS_IDLE 1
#define portSUPPRESS_TICKS_AND_SLEEP( xExpectedIdleTime ) vApplicationSleep( xExpectedIdleTime )

void vApplicationSleep( TickType_t xExpectedIdleTime ) {
  Serial.println("I can go to sleep");
}

void loop() {

}

void TaskPrint(void *pvParameters)
{
  (void) pvParameters;

  for (;;)
  {
    Serial.println("Running");
    vTaskDelay( 10000 / portTICK_PERIOD_MS );
  }
}
```

I expect vApplicationSleep to be called whenever TaskPrint is not running, that is not the case however. It seems like the define statements are being overwritten. I've have tried moving them to FreeRTOSConfig.h but that caused a compile error as tasks.c was unable to find my function (vApplicationSleep).

Im using the Arduino IDE and a AtMega32u4

**Comment by feilipu** [comment]
Hi tim-vu. The tickless mode is not supported in the (this) AVR port. As you note, there isn't a standard `vApplicationSleep()` function. If you were to write one you'd need to stop the watchdog timer as part of entering this mode.

What is happening is that the Arduino `loop()` function is being called by the [idle Task Hook](https://github.com/feilipu/Arduino_FreeRTOS_Library/blob/master/src/variantHooks.cpp#L73). And this idle hook function will always keep the scheduler busy.

It is possible to select a longer tick interval. And sleep following each tick.

Or alternatively put the AVR to sleep once its tasks have been done. You can do this in the loop function, as described [here](https://feilipu.me/2015/11/24/arduino_freertos/).
